Library/PaginateTests.swift

```diff
@@ -236,6 +240,116 @@ final class PaginateTests: TestCase {
     )
   }
 
+  func testPaginateFlow_combine() {
+    let newRequest_combine = PassthroughSubject<Int, Never>()
+    let nextPage_combine = PassthroughSubject<(), Never>()
+    let requestFromParams_combine: (Int) -> any Publisher<[Int], ConcreteError> = { p in
+      Just([p]).setFailureType(to: ConcreteError.self)
+    }
+    let requestFromCursor_combine: (Int) -> any Publisher<[Int], ConcreteError> = { c in
+      Just(c <= 2 ? [c] : []).setFailureType(to: ConcreteError.self)
+    }
+
+    let (values, loading, _, _) = paginate_combine(
+      requestFirstPageWith: newRequest_combine,
+      requestNextPageWhen: nextPage_combine,
+      clearOnNewRequest: true,
+      valuesFromEnvelope: valuesFromEnvelope,
+      cursorFromEnvelope: cursorFromEnvelope,
+      requestFromParams: requestFromParams_combine,
+      requestFromCursor: requestFromCursor_combine
+    )
+
+    let valuesTest = CombineTestObserver<[Int], Never>()
+    valuesTest.observe(values)
+
+    let loadingTest = CombineTestObserver<Bool, Never>()
+    loadingTest.observe(loading)
+
+    valuesTest.assertDidNotEmitValue("No values emit immediately.")
+    valuesTest.assertDidNotFinish()
+    loadingTest.assertDidNotEmitValue("No loading happens immediately.")
+    loadingTest.assertDidNotFinish()
```

Comment:
My goal here was to do this TDD-style. The original `paginate` method includes a lot of code that isn't 1:1 mappable to Combine, like `switchMap` and `Either`. I think the only way I'm going to be able to re-write it is to go feature-by-feature with the test, and make sure the behavior is the same. 

So the combine test is _literally_ copy-pasted from the original test; I've been adding methods to `CombineTestObserver` so that it can work in the same way.